include/git2/errors.h

```diff
@@ -58,6 +60,14 @@ typedef enum {
 	GIT_EMISMATCH       = -33,	/**< Hashsum mismatch in object */
 	GIT_EINDEXDIRTY     = -34,	/**< Unsaved changes in the index would be overwritten */
 	GIT_EAPPLYFAIL      = -35,	/**< Patch application failed */
+	GIT_EAGAIN          = -36,	/**< Operation would block */
+	GIT_ECANCEL         = -37,	/**< Operation was cancelled by application */
```

Comment:
How is `GIT_ECANCEL` different from `GIT_EUSER`?

Reply:
The documentation for GIT_EUSER says:
> GIT_EUSER is a special error that is never generated by libgit2
> code.  You can return it from a callback (e.g to stop an iteration)
> to know that it was generated by the callback and not by libgit2.

However, if you look at src/transports/smart_protocol.c, you can see that GIT_EUSER is in fact generated by libgit2 after a call to git_remote_stop().
This contradicts documented behaviour. Semantically, yes, the operation is cancelled by the user and in this way returning GIT_EUSER may make some sense, but the user never returned this error code from any callback so this behaviour is confusing.

Reply:
Indeed, we should clarify this in the documentation.